src/Cohesity.Powershell/Cmdlets/ProtectionSource/GetCohesityProtectionSourceObject.cs

```diff
@@ -165,12 +165,44 @@ private IEnumerable<ProtectionSourceNode> FlattenNodes(IEnumerable<ProtectionSou
             foreach(var node in nodes)
             {
                 var childrenNodes = node.Nodes;
+                // for SQL nodes, the items are available in ApplicationNodes
+                if (childrenNodes == null)
+                {
+                    if(node.ApplicationNodes != null)
+                    {
+                        childrenNodes = node.ApplicationNodes;
+                    }
+                }
                 node.Nodes = null;
-                result.Add(node);
+
+                DecideToAppendNode(node, ref result);
+
                 result.AddRange(FlattenNodes(childrenNodes));
             }
 
             return result;
         }
+        private void DecideToAppendNode(ProtectionSourceNode node, ref List<ProtectionSourceNode> result)
+        {
+            // Ensure that we are processing only the requested items type passed in the environment variable
+            Model.ProtectionSource.EnvironmentEnum resp = System.Array.Find(this.Environments, element => element == node.ProtectionSource.Environment);
```

Comment:
wrap around the line length, whats it for powershell ?

Reply:
Done